src/dataflow/src/server/boundary.rs

```diff
@@ -116,33 +124,41 @@ mod event_link {
     impl ComputeReplay for EventLinkBoundary {
         fn replay<G: Scope<Timestamp = mz_repr::Timestamp>>(
             &mut self,
-            id: GlobalId,
+            id: SourceInstanceKey,
             scope: &mut G,
             name: &str,
-        ) -> Option<(
+        ) -> (
             Collection<G, Row, Diff>,
             Collection<G, DataflowError, Diff>,
             Rc<dyn Any>,
-        )> {
-            self.sources.remove(&id).map(|source| {
-                let ok = Some(source.ok.inner)
-                    .mz_replay(
-                        scope,
-                        &format!("{name}-ok"),
-                        Duration::MAX,
-                        source.ok.activator,
-                    )
-                    .as_collection();
-                let err = Some(source.err.inner)
-                    .mz_replay(
-                        scope,
-                        &format!("{name}-err"),
-                        Duration::MAX,
-                        source.err.activator,
-                    )
-                    .as_collection();
-                (ok, err, source.token)
-            })
+        ) {
+            // Ensure that a channel pair exists.
+            if !self.send.contains_key(&id) {
+                let (send, recv) = crossbeam_channel::unbounded();
+                self.send.insert(id.clone(), send);
+                self.recv.insert(id.clone(), recv);
+            }
+
+            let source = self.recv[&id].recv().expect("Unable to acquire source");
```

Comment:
Yeah, the goal here is to *block* until available, because the source may not be ready at the moment we call. In a better future world, we would prepare this async, and then deploy it when ready (in the same construction order across workers, though). The crashing here is if the channel gets severed, and .. yeah that should be handled with some retry / idk, but it seems like a "unrecoverable error" if we can't get a channel, and crashing seems good for now.